src/include/Parser.h

```diff
@@ -7,23 +7,17 @@
 
 class Parser {
     Lexer m_lexer;
-    Token m_currentToken;
-
 public:
     explicit Parser(Lexer lexer);
-    void buildTree();
+    std::unique_ptr<AST::Stmt> parse();
 private:
-    Token getNextToken() {
-        return m_currentToken = m_lexer.getToken();
-    }
     // Parsers
-    std::unique_ptr<AST::Stmt> parseStatement();
-    // Expressions
-    std::unique_ptr<AST::Expr> parseExpression();
-    std::unique_ptr<AST::Expr> parseLiteral();
-    // Basic statements
-    std::unique_ptr<AST::FnDecl> parseFunction();
-    std::unique_ptr<AST::ReturnStmt> parseReturn();
+    Token m_currenttoken;
+    Token eattoken(TokenType t);
```

Comment:
Also we should perhaps keep in mind `const` correctness; for instance `t` here can be `const`. In this case it doesn't really matter but it is a best practice.